// This file is auto-generated, DO NOT make any changes here
syntax = "proto3";
package app.coronawarn.server.common.protocols.internal.pt;
option java_multiple_files = true;

message CheckInRecord {
  uint32 startIntervalNumber = 1;
  uint32 period = 2;
  uint32 transmissionRiskLevel = 3;
}

message CheckInProtectedReport {
  bytes locationIdHash = 1;
  bytes iv = 2;
  bytes encryptedCheckInRecord = 3;
  bytes mac = 4;
}

message TraceWarningPackage {
  // hours since UNIX Epoch
  uint32 intervalNumber = 1;
  string region = 2;
  repeated TraceTimeIntervalWarning timeIntervalWarnings = 3  [deprecated = true];
  repeated CheckInProtectedReport checkInProtectedReports = 4;
}

message TraceTimeIntervalWarning {
  bytes locationIdHash = 1; // SHA-256 of the Location ID
  // 10-minute intervals since UNIX Epoch
  uint32 startIntervalNumber = 2;
  // Number of 10-minute intervals to which the warning applies
  uint32 period = 3;
  uint32 transmissionRiskLevel = 4;
}
